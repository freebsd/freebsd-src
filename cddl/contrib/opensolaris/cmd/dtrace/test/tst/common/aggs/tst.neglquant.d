/*
 * CDDL HEADER START
 *
 * The contents of this file are subject to the terms of the
 * Common Development and Distribution License (the "License").
 * You may not use this file except in compliance with the License.
 *
 * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
 * or http://www.opensolaris.org/os/licensing.
 * See the License for the specific language governing permissions
 * and limitations under the License.
 *
 * When distributing Covered Code, include this CDDL HEADER in each
 * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
 * If applicable, add the following below this CDDL HEADER, with the
 * fields enclosed by brackets "[]" replaced with your own identifying
 * information: Portions Copyright [yyyy] [name of copyright owner]
 *
 * CDDL HEADER END
 */

/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 * Use is subject to license terms.
 */

#pragma ident	"%Z%%M%	%I%	%E% SMI"

#pragma D option quiet

BEGIN
{
	@["j-church"] = lquantize(1, 0, 10, 1, 100);
	@["j-church"] = lquantize(1, 0, 10, 1, -99);
	@["j-church"] = lquantize(1, 0, 10, 1, -1);
	val = 123;
}

BEGIN
{
	@["k-ingleside"] = lquantize(1, 0, 10, 1, -val);
}

BEGIN
{
	@["l-taraval"] = lquantize(0, 0, 10, 1, -val);
	@["l-taraval"] = lquantize(-1, 0, 10, 1, -val);
	@["l-taraval"] = lquantize(1, 0, 10, 1, val);
	@["l-taraval"] = lquantize(1, 0, 10, 1, val);
}

BEGIN
{
	@["m-oceanview"] = lquantize(1, 0, 10, 1, (1 << 63) - 1);
	@["m-oceanview"] = lquantize(1, 0, 10, 1);
	@["m-oceanview"] = lquantize(2, 0, 10, 1, (1 << 63) - 1);
	@["m-oceanview"] = lquantize(8, 0, 10, 1, 400000);
}

BEGIN
{
	@["n-judah"] = lquantize(1, 0, 10, 1, val);
	@["n-judah"] = lquantize(2, 0, 10, 1, val);
	@["n-judah"] = lquantize(2, 0, 10, 1, val);
	@["n-judah"] = lquantize(2, 0, 10, 1);
}

BEGIN
{
	this->i = 1;
	this->val = (1 << 63) - 1;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["f-market"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;
}

BEGIN
{
	this->i = 1;

	/*
	 * We want to test the ability to sort very large quantizations
	 * that differ by a small amount.  Ideally, they would differ only
	 * by 1 -- but that is smaller than the precision of long doubles of
	 * this magnitude on x86.  To assure that the same test works on x86
	 * just as it does on SPARC, we pick a value that is just larger than
	 * the precision at this magnitude.  It should go without saying that
	 * this robustness on new ISAs very much depends on the precision
	 * of the long double representation.
	 */
	this->val = (1 << 63) - 7;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;

	@["s-castro"] = lquantize(this->i, 0, 10, 1, this->val);
	this->i++;
	this->val = ((1 << 63) - 1) / this->i;
}

BEGIN
{
	exit(0);
}
