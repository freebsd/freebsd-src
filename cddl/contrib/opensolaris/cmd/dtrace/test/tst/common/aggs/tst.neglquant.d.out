
  k-ingleside                                       
           value  ------------- Distribution ------------- count    
               0                                         | 0        
               1 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@| -123     
               2                                         | 0        

  j-church                                          
           value  ------------- Distribution ------------- count    
             < 0 |                                         0        
               0 |                                         0        
               1 |                                         0        

  l-taraval                                         
           value  ------------- Distribution ------------- count    
             < 0                @@@@@|                     -123     
               0                @@@@@|                     -123     
               1                     |@@@@@@@@@@           246      
               2                     |                     0        

  n-judah                                           
           value  ------------- Distribution ------------- count    
               0 |                                         0        
               1 |@@@@@@@@@@@@@                            123      
               2 |@@@@@@@@@@@@@@@@@@@@@@@@@@@              247      
               3 |                                         0        

  m-oceanview                                       
           value  ------------- Distribution ------------- count    
               0                     |                     0        
               1           @@@@@@@@@@|                     -9223372036854775808
               2                     |@@@@@@@@@@           9223372036854775807
               3                     |                     0        
               4                     |                     0        
               5                     |                     0        
               6                     |                     0        
               7                     |                     0        
               8                     |                     400000   
               9                     |                     0        

  s-castro                                          
           value  ------------- Distribution ------------- count    
               0 |                                         0        
               1 |@@@@@@@@@@@@@@@                          9223372036854775801
               2 |@@@@@@@@                                 4611686018427387903
               3 |@@@@@                                    3074457345618258602
               4 |@@@@                                     2305843009213693951
               5 |@@@                                      1844674407370955161
               6 |@@@                                      1537228672809129301
               7 |@@                                       1317624576693539401
               8 |                                         0        

  f-market                                          
           value  ------------- Distribution ------------- count    
               0 |                                         0        
               1 |@@@@@@@@@@@@@@@                          9223372036854775807
               2 |@@@@@@@@                                 4611686018427387903
               3 |@@@@@                                    3074457345618258602
               4 |@@@@                                     2305843009213693951
               5 |@@@                                      1844674407370955161
               6 |@@@                                      1537228672809129301
               7 |@@                                       1317624576693539401
               8 |                                         0        

